standards/application/messaging-api.md

```diff
@@ -0,0 +1,965 @@
+- a [LIGHTPUSH](../standards/core/lightpush.md) endpoint for broadcasting data;
+- a [FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md) endpoint for receiving circulating data.
+
+If not provided, nodes discovered through the network SHOULD be used.
+
+##### `bootstrapNodes`
+This property MUST be provided.
+
+A list of `Multiaddr` addresses to remote peers that SHOULD be used for any applicable method of discovery that MAY include:
+- [EIP-1459](https://eips.ethereum.org/EIPS/eip-1459);
+- [DISCV5](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/standards/core/33/discv5.md);
+- [PEER-EXCHANGE](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/standards/core/34/peer-exchange.md).
+
+##### `confirmStored`
+An optional property that defaults to `false`.
+If set to `true` and `confirmContentTopics` are provided, a recurring background [STORE](../standards/core/store.md) query (as described in the `Message Storage API` section) MUST be initiated.
+If set to `true` without `confirmContentTopics`, the background [STORE](../standards/core/store.md) query (as described in the `Messaging Storage API` section) MUST only be initiated after the `Subscribe API` is called.
+If set to `false`, the `stored` property on the `MessageRecord` (defined in the `Message Storage API`) MUST NOT be populated, unless `History API` is triggered.
+
+##### `confirmReceived`
+An optional property that defaults to `true`.
+If set to `true`, the node MUST initiate a network listener for messages circulated under `confirmContentTopics` via either [FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md) or [RELAY](https://github.com/vacp2p/rfc-index/blob/0277fd0c4dbd907dfb2f0c28b6cde94a335e1fae/waku/standards/core/11/relay.md) using the `Subscribe API`.
+If set to `true` but `confirmContentTopics` are not provided, the received property on the `MessageRecord` (as defined in the `Message Storage API`) MUST be populated only after the `Subscribe API` is called.
+If set to `false`, the received property MUST NOT be populated, unless `Subscribe API` is called.
+
+##### `confirmContentTopics`
+An optional property that SHOULD be provided in conjunction with either `confirmStored` or `confirmReceived`.
+It includes an array of `contentTopics` that the node MUST start monitoring in the background upon initialization,
+as defined in the [TOPICS](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/informational/23/topics.md#content-topic-format) specification.
+
+#### Programmatic API / REST API
+
+No methods or `REST` endpoints are provided for initial configuration.
+Instead, the initial configuration MUST be supplied at node creation time according to the guidelines of the chosen implementation.
+
+### Send
```

Comment:
SHould it also mention what happens when the node is in `Unhealthy` state? Or is that implicit - sending will fail and it will be retried when the health changes to `minimally healthy` or `healthy`?

Reply:
it's specified in `Retry` section below, if it is not clear, please, re-open this comment